v8/types/Authenticator.go

```diff
@@ -79,3 +108,22 @@ func (a *Authenticator) Marshal() ([]byte, error) {
 	b = asn1tools.AddASNAppTag(b, asnAppTag.Authenticator)
 	return b, nil
 }
+
+func (c *CredDelegation) Unmarshal(b []byte) error {
+	c.BndLength = binary.LittleEndian.Uint32(b[0:4])
+	if c.BndLength != 16 {
+		return fmt.Errorf("Invalid BndLength")
```

Comment:
Consider adding the length here for easier debugging.

Reply:
Fixed.